### PR TITLE
[`SPM`] Patch `spm` Llama and T5

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
An edge case was found, so we need to find a more reliable way to make sure we don't strip the first token, but also have the correct length. 

Previously, we would strip `>` from a tokenization if `f">>"` exists in the vocab  ( and `unk_token = <unk>`). 

Before:
```python 
tokenizer.tokenize("Hey <s>>")
['▁Hey', '▁', '<s>']
```
after
```python 
tokenizer.tokenize("<s>>")
['▁Hey', '▁', '<s>', '>']
```
The new logic makes more sense too. 

# Why was this not caught before? 
Because the common spm tests only use the `sample_vocab` and since it is an edge case, did not appear in the different integration tests. 

# TODO
Add more and more edge cases.... Let's make sure this is the last one. Will add a pr for tests, cc @sgugger

### CODE REVIEW

**Comment by ArthurZucker** [comment]
TODO is just adding the test that was missing!

File: tests/models/llama/test_tokenization_llama.py [hunk 1]
```diff
@@ -546,6 +546,15 @@ def test_special_token_special_word(self):
         decoded_tokens = tokenizer.decode(input_ids)
         self.assertEqual(decoded_tokens, " <s> Hello<s> how")
 
+    def test_some_edge_cases(self):
```
Comment (faabian) [inline tests/models/llama/test_tokenization_llama.py:549]: I suggest another edge case with the string ">". I'm still afraid it will tokenize as `["<", "unk", ">>"]` and then cut off the leading three elements, resulting in `[]` (?).

**Review by sgugger — APPROVED** [review]
Thanks!

File: tests/models/llama/test_tokenization_llama.py (hunk 1 above)
Comment (ArthurZucker) [reply tests/models/llama/test_tokenization_llama.py:549]: The string `'>'` does not start with `SPIECE_UNDERLINE` so will just be tokenized with the sp model to `>`. 
Wil add it too 😉
